crates/http/src/lib.rs

```diff
@@ -59,43 +54,56 @@ pub struct HttpTrigger {
     engine: Arc<ExecutionContext>,
 }
 
-impl HttpTrigger {
-    /// Creates a new Spin HTTP trigger.
-    pub async fn new(
-        builder: Builder<SpinHttpData>,
-        app: Application<CoreComponent>,
-        address: String,
-        tls: Option<TlsConfig>,
-    ) -> Result<Self> {
-        let trigger_config = app
-            .info
-            .trigger
-            .as_http()
-            .ok_or_else(|| anyhow!("Application trigger is not HTTP"))?
-            .clone();
-
-        let component_triggers = app.component_triggers.try_map_values(|id, trigger| {
-            trigger
-                .as_http()
-                .cloned()
-                .ok_or_else(|| anyhow!("Expected HTTP configuration for component {}", id))
-        })?;
+#[derive(Clone)]
+pub struct HttpRuntimeConfig {
```

Comment:
Nit: Anything *but* just "Runtime"......maybe _HttpTriggerRuntimeConfig_? ......pretty please

Reply:
OR *HttpTriggerConfig*

Reply:
Whatya think?

Reply:
there are just too many configs...
1. `HttpTriggerConfiguration`
2. `HttpConfig`
3. and this `HttpRuntimeConfig`

How about `HttpRuntimeConfig` -> `HttpTriggerExecutionConfig`?

Reply:
This might be one for a subsequent blitz on naming - I don't even know the difference between the first two let along the new one - we should do a pass for figure out their different responsibilities and give them clearly distinct names that clearly express those differences, but this PR might not be the place for that, not sure.

Reply:
Agreed. My understanding is that

1. The first one is for top level trigger struct `"trigger = { type = "http", base = "/" }`
2. The second one is for component trigger struct 
  ```
  [component.trigger]
  route = "/hello"
  ```
3. The third one is for CLI aguments: http has `--listen address`, `--log-dir` and `--tls-cert/key`

Reply:
I agree that we shouldn't try to name everything perfectly in this PR. The names were already pretty confusing.